src/test/java/uk/gov/hmcts/reform/adoption/service/CoreCaseDataServiceTest.java

```diff
@@ -0,0 +1,76 @@
+package uk.gov.hmcts.reform.adoption.service;
+
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+import uk.gov.hmcts.reform.adoption.request.RequestData;
+import uk.gov.hmcts.reform.authorisation.generators.AuthTokenGenerator;
+import uk.gov.hmcts.reform.ccd.client.CoreCaseDataApi;
+import uk.gov.hmcts.reform.ccd.client.model.CaseDataContent;
+import uk.gov.hmcts.reform.ccd.client.model.Event;
+import uk.gov.hmcts.reform.ccd.client.model.StartEventResponse;
+import uk.gov.hmcts.reform.idam.client.IdamClient;
+import uk.gov.hmcts.reform.idam.client.models.UserDetails;
+
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+import static uk.gov.hmcts.reform.adoption.service.CoreCaseDataService.CASE_TYPE;
+import static uk.gov.hmcts.reform.adoption.service.CoreCaseDataService.JURISDICTION;
+import static uk.gov.hmcts.reform.adoption.service.CoreCaseDataService.START_CASE_EVENT;
+
+@ExtendWith(MockitoExtension.class)
+class CoreCaseDataServiceTest extends BaseTest {
+
+    @Mock
+    private CoreCaseDataApi coreCaseDataApi;
+
+    @Mock
+    private AuthTokenGenerator authTokenGenerator;
+
+    @Mock
+    private IdamClient idamClient;
+
+    @Mock
+    private RequestData requestData;
+
+    @InjectMocks
+    private CoreCaseDataService service;
+
+    @BeforeEach
+    void setup() {
+        when(authTokenGenerator.generate()).thenReturn(serviceAuthToken);
+        when(idamClient.getUserDetails(userAuthToken)).thenReturn(UserDetails.builder().id(userId).build());
+        when(requestData.authorisation()).thenReturn(userAuthToken);
+    }
+
+    @Test
+    @DisplayName("Should return a new case")
+    void returnNewCase() {
+        when(coreCaseDataApi.startCase(userAuthToken, serviceAuthToken, CASE_TYPE, START_CASE_EVENT))
```

Comment:
personally i believe that test should not use constant from code like CASE_TYPE, if somebody accidentally or for some reason change case type, then this test should catch it

Reply:
these tests are independent of the definition files though, so I am not sure of the benefit of this

Reply:
It is independent, but we know that e.g. jurisdiction must be ADOPTION. If somebody changes constant in CoreCaseDataService breaking this contract, the test wont spot it